accelerator/common_core.c

```diff
@@ -450,3 +454,79 @@ status_t ta_send_bundle(const iota_config_t* const tangle, const iota_client_ser
 
   return SC_OK;
 }
+
+status_t ta_receive_mam_message(const iota_client_service_t* const service, const char* const chid, char** payload) {
+  status_t ret = SC_OK;
+  mam_api_t mam;
+  tryte_t* payload_trytes = NULL;
+  size_t payload_size = 0;
+  bundle_transactions_t* bundle = NULL;
+  bundle_transactions_new(&bundle);
+  bool is_last_packet;
+  tryte_t bundle_hash[NUM_TRYTES_BUNDLE];
+  find_transactions_req_t* txn_req = find_transactions_req_new();
+  find_transactions_res_t* txn_res = find_transactions_res_new();
+  ta_find_transaction_objects_req_t* obj_req = ta_find_transaction_objects_req_new();
+  transaction_array_t* obj_res = transaction_array_new();
+
+  // Creating MAM API
+  if (mam_api_init(&mam, (tryte_t*)SEED) != RC_OK) {
```

Comment:
Sorry, maybe I'm not clear enough. These API should under map library and there should be wrapper for them. all accelerator files should all handle logic of tangle protocol only.